oasis-build/src/visitor/hir.rs

```diff
@@ -168,26 +179,36 @@ impl<'tcx> hir::intravisit::Visitor<'tcx> for EventCollector<'tcx> {
             _ => None,
         };
         if let Some(emit_arg) = emit_arg {
-            let emit_arg_ty = self
-                .tcx
-                .typeck_tables_of(emit_arg.hir_id.owner_def_id())
-                .expr_ty(&emit_arg);
-            if let Some(adt_def) = emit_arg_ty.ty_adt_def().or_else(|| match emit_arg_ty.sty {
-                ty::TyKind::Ref(
-                    _,
-                    TyS {
-                        sty: ty::TyKind::Adt(adt_def, _),
-                        ..
-                    },
-                    _,
-                ) => Some(adt_def),
-                _ => None,
-            }) {
-                self.adt_defs
-                    .entry(adt_def)
-                    .or_default()
-                    .push(emit_arg.span);
-            };
+            let did = emit_arg.hir_id.owner_def_id();
+            let emit_arg_ty = self.tcx.typeck_tables_of(did).expr_ty(&emit_arg);
+            macro_rules! insert_def_ty {
```

Comment:
is the macro really required here? Or could you just inline the code within the match statement?

Reply:
I use it twice and have to inline it because it takes a mut ref of `self.tcx` but I need an immutable ref in the `match`. I'd rather not copy+paste code.

Reply:
got I didn't see the second use of the macro 👍